projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java

```diff
@@ -124,6 +118,23 @@ public boolean buildImage(
       String imageName, DataJob dataJob, JobDeployment jobDeployment, Boolean sendNotification)
       throws ApiException, IOException, InterruptedException {
 
+    String builderAwsSecretAccessKey = awsCredentialsService.getAwsSecretAccessKey();
+    String builderAwsAccessKeyId = awsCredentialsService.getAwsAccessKeyId();
+    String builderAwsSessionToken = "";
+    String awsRegion = awsCredentialsService.getAwsRegion();
+
+    if (awsCredentialsService.isAssumeIAMRole()) {
+      // Iff temporary credentials flag is enabled we generate role credentials
+      // to use when authenticating against ECR
+      var credentials = awsCredentialsService.createTemporaryCredentials();
+      builderAwsSecretAccessKey =
+          credentials.get(AWSCredentialsService.AWS_SERVICE_ACCOUNT_SECRET_ACCESS_KEY);
+      builderAwsAccessKeyId =
+          credentials.get(AWSCredentialsService.AWS_SERVICE_ACCOUNT_ACCESS_KEY_ID);
+      builderAwsSessionToken =
+          credentials.get(AWSCredentialsService.AWS_SERVICE_ACCOUNT_SESSION_TOKEN);
+    }
+
```

Comment:
Isn't it better to put this logic in AwsCredentialsService that is responsible for building AWS credentials. For example method called "getCredentials" would return the correct ones depending on the flag